src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenFunctionPointersTests.cs

```diff
@@ -4051,6 +4077,169 @@ .maxstack  3
+    {
+        delegate*<void> ptr = default;
+        Console.Write(ptr is null);
+    }
+}", expectedOutput: "True");
+
+            verifier.VerifyIL("C.Main", @"
+{
+  // Code size       11 (0xb)
+  .maxstack  2
+  IL_0000:  ldc.i4.0
+  IL_0001:  conv.u
+  IL_0002:  ldnull
+  IL_0003:  ceq
+  IL_0005:  call       ""void System.Console.Write(bool)""
+  IL_000a:  ret
+}
+");
+        }
+
+        [Fact]
+        public void ParamsArrayOfFunctionPointers()
+        {
+            var verifier = CompileAndVerifyFunctionPointers(@"
+unsafe class C
+{
+    static void Params(params delegate*<void>[] funcs)
+    {
+        foreach (var f in funcs)
+        {
+            f();
+        }
+    }
+
+    static void Main()
+    {
+        Params();
+    }
+}", expectedOutput: "");
+
+            verifier.VerifyIL("C.Main", expectedIL: @"
+{
+  // Code size       12 (0xc)
+  .maxstack  1
+  IL_0000:  ldc.i4.0
+  IL_0001:  newarr     ""delegate*<void>""
+  IL_0006:  call       ""void C.Params(params delegate*<void>[])""
+  IL_000b:  ret
+}
+");
+        }
+
+        [Fact]
+        public void SpanOfFunctionPointers()
+        {
+            var verifier = CompileAndVerifyFunctionPointers(@"
+using System;
+static unsafe class C
+{
+    static int Getter(int i) => i;
+    static void Print(delegate*<int, int>* p)
+    {
+        for (int i = 0; i < 3; i++)
+            Console.Write(p[i](i));
+    }
+
+    static void Main()
+    {
+        delegate*<int, int>* p = stackalloc delegate*<int, int>[] { &Getter, &Getter, &Getter };
+        Print(p);
+    }
+}
+", expectedOutput: "012");
+
+            verifier.VerifyIL("C.Main", expectedIL: @"
+{
+  // Code size       58 (0x3a)
+  .maxstack  4
+  IL_0000:  ldc.i4.3
+  IL_0001:  conv.u
+  IL_0002:  sizeof     ""delegate*<int,int>""
+  IL_0008:  mul.ovf.un
+  IL_0009:  localloc
+  IL_000b:  dup
+  IL_000c:  ldftn      ""int C.Getter(int)""
+  IL_0012:  stind.i
+  IL_0013:  dup
+  IL_0014:  sizeof     ""delegate*<int,int>""
+  IL_001a:  add
+  IL_001b:  ldftn      ""int C.Getter(int)""
+  IL_0021:  stind.i
+  IL_0022:  dup
+  IL_0023:  ldc.i4.2
+  IL_0024:  conv.i
+  IL_0025:  sizeof     ""delegate*<int,int>""
+  IL_002b:  mul
+  IL_002c:  add
+  IL_002d:  ldftn      ""int C.Getter(int)""
+  IL_0033:  stind.i
+  IL_0034:  call       ""void C.Print(delegate*<int,int>*)""
+  IL_0039:  ret
+}
+");
+        }
+
+        [Fact]
+        public void RecusivelyUsedTypeInFunctionPointer()
+        {
+            var verifier = CompileAndVerifyFunctionPointers(@"
+namespace Interop
+{
+    public unsafe struct PROPVARIANT
+    {
+        public CAPROPVARIANT ca;
+    }
+    public unsafe struct CAPROPVARIANT
+    {
+        public uint cElems;
+        public delegate*<PROPVARIANT> pElems;
+        public delegate*<PROPVARIANT> pElemsProp { get; }
+    }
+}");
+        }
+
+        [Fact]
+        public void VolatileFunctionPointerField()
+        {
+            var verifier = CompileAndVerifyFunctionPointers(@"
```

Comment:
>CompileAndVerifyFunctionPointers [](start = 27, length = 32)

Consider asserting the output.

Reply:
Whoops, I meant to. Will add.

---
In reply to: [407749446](https://github.com/dotnet/roslyn/pull/43248#discussion_r407749446) [](ancestors = 407749446)